File: hipstore/src/main/java/tech/lab23/hipstore/Hipstore.java
package tech.lab23.hipstore;

import java.util.List;

interface Hipstore {
    interface MultiEntities<T> {
        boolean contains(T item);
        void remove(T item);
        void add(T item);
        void add(T item, int index);
        void addAll(List<T> items);
        void addAll(int position, List<T> items);
        List<T> getAll();
        void clear();
    }

    interface SingleEntity<T> {
        void remove(T item);
        void put(T item);
        T get();
    }
}

File: hipstore/src/main/java/tech/lab23/hipstore/Hipstore.kt
package tech.lab23.hipstore

import java.io.Serializable

interface Hipstore {

    interface MultiEntities<T : Serializable> {
        fun contains(item: T): Boolean
        fun remove(item: T)
        fun add(item: T)
        fun getAll(): MutableList<T>
        fun clear()
    }

    interface SingleEntity<T : Serializable> {
        fun remove(item: T)
        fun put(item: T)
        fun get(): T
    }

}